plugins/rtmp-services/rtmp-common.c

```diff
@@ -531,6 +536,48 @@ static void update_protocol(json_t *service, obs_data_t *settings)
 	obs_data_set_string(settings, "protocol", "RTMP");
 }
 
+static void copy_info_to_settings(json_t *service, obs_data_t *settings)
+{
+	const char *name = obs_data_get_string(settings, "service");
+
+	fill_more_info_link(service, settings);
+	fill_stream_key_link(service, settings);
+	copy_string_from_json_if_available(service, settings,
+					   "ertmp_configuration_url");
+	copy_string_from_json_if_available(service, settings,
+					   "ertmp_multitrack_video_name");
+	if (!obs_data_has_user_value(settings, "ertmp_multitrack_video_name")) {
+		obs_data_set_string(settings, "ertmp_multitrack_video_name",
+				    "Multitrack Video");
+	}
+
+	const char *learn_more_link_url =
+		get_string_val(service, "ertmp_configuration_learn_more_link");
+	struct dstr learn_more_link = {0};
+	if (learn_more_link_url) {
+		dstr_init_copy(
+			&learn_more_link,
+			obs_module_text("MultitrackVideo.LearnMoreLink"));
+		dstr_replace(&learn_more_link, "%1", learn_more_link_url);
+	}
+
+	struct dstr str;
+	dstr_init_copy(&str, obs_module_text("MultitrackVideo.Disclaimer"));
+	dstr_replace(&str, "%1",
+		     obs_data_get_string(settings,
+					 "ertmp_multitrack_video_name"));
+	dstr_replace(&str, "%2", name);
+	if (learn_more_link.array) {
+		dstr_replace(&str, "%3", learn_more_link.array);
```

Comment:
Instead of `dstr_replace`, could this not be `dstr_cat` or `dstr_catf` or something? I would think that, if a service had not specified `learn_more_link`, the string would have a lingering `%3` in it on user display.

Reply:
oh, yes, you're right; at the very least the `%3` needs to be removed when `learn_more_link` is empty

Reply:
changed in https://github.com/obsproject/obs-studio/pull/10543/commits/9608ddfc056bd2d21bce092ce736961acea0eff0

`dstr_catf` seems tempting, but I'd rather not pass external strings as format arg to `printf`-like functions

Reply:
Thanks. Resolving.